27_tictactoe/lib/board.py

```diff
@@ -0,0 +1,79 @@
+from lib.player import Player
+from random import randint
+
+"""
+start(p1, p2) - starts a new game with players p1 and p2
+
+play(row,column) - current player picked a square. If the square was free mark it as X or O and switch turns to the other player.
+If square was taken or not in the board, or game is over, raise an exception.
+
+winner() - If game is over return the winning player, else return None
+
+val(row, column) - return current value for square (X, O or empty)
+"""
+
+class InvalidMove(Exception):
+    def __init__(self,msg):
+        self.message = "Invalid move in a tick-tack-toe; %s" %(msg)
+
+
+class Game:
+    def __init__(self):
+        self.reset_board()
+        self.players = [None,None]
+        self.current_player = None
+        self.winner = None
+        self.moves_counter = 0
+    
+    def reset_board(self):
+        self.data = [ [None,None,None] , 

```

Comment:
This short writing leads to a bug in the program since this structure behaves differently.
Do you happen to know the reason?

I opened an issue for it: https://github.com/ynonp/python-examples-verint-2016-07/issues/222#issue-168286861

Thanks, Nir

From: ynonp [mailto:notifications@github.com]
Sent: Friday, July 29, 2016 8:39 AM
To: ynonp/python-examples-verint-2016-07
Cc: Naor, Nir; Author
Subject: Re: [ynonp/python-examples-verint-2016-07] ToCode Solution: תרגיל מסכם מחלקות (#221)


In 27_tictactoe/lib/board.py<https://github.com/ynonp/python-examples-verint-2016-07/pull/221#discussion_r72743693>:

> +

> +class InvalidMove(Exception):

> +    def __init__(self,msg):

> +        self.message = "Invalid move in a tick-tack-toe; %s" %(msg)

> +

> +

> +class Game:

> +    def __init__(self):

> +        self.reset_board()

> +        self.players = [None,None]

> +        self.current_player = None

> +        self.winner = None

> +        self.moves_counter = 0

> +

> +    def reset_board(self):

> +        self.data = [ [None,None,None] ,

Python has * operator for lists, so this also works: self.data = [[None] * 3] * 3

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ynonp/python-examples-verint-2016-07/pull/221/files/e7ea17bf98533b25fcd24aa58240127dd7112296#r72743693>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ATWluudmApEs6G91PoGk9rXOIHqbyV8gks5qaZIEgaJpZM4JXyZi>.

-- 
This electronic message may contain proprietary and confidential 
information of Verint Systems Inc., its affiliates and/or subsidiaries.
The information is intended to be for the use of the individual(s) or
entity(ies) named above. If you are not the intended recipient (or 
authorized to receive this e-mail for the intended recipient), you may not 
use, copy, disclose or distribute to anyone this message or any information 
contained in this message. If you have received this electronic message in 
error, please notify us by replying to this e-mail.